.travis.yml

```diff
@@ -36,11 +29,24 @@ cache:
   - "$HOME/.gradle/wrapper/"
 before_install:
 - ./travis/prepare-signing.sh $encrypted_677f232983c0_key $encrypted_677f232983c0_iv
-- if [ "$CROSSDOCK" == true ]; then bash ./travis/install-crossdock-deps.sh ; fi
-script:
-- if [ "$TESTS" == true ]; then make test-travis ; else echo 'skipping tests'; fi
-- if [ "$COVERAGE" == true ]; then ./gradlew codeCoverageReport ; else echo 'skipping coverage'; fi
-- if [ "$CROSSDOCK" == true ]; then bash ./travis/build-crossdock.sh ; else echo 'skipping crossdock'; fi
-after_success:
-- if [ "$COVERAGE" == true ]; then bash <(curl -s https://codecov.io/bash) ; fi
-- if [ "$CROSSDOCK" == true -a "$TRAVIS_PULL_REQUEST" == false ]; then ./gradlew upload ; fi
+
+stages:
+  - test
+  - name: deploy
+    if: env(TRAVIS_PULL_REQUEST) = false
+
+jobs:
+  include:
+    - stage: test
+      env: [ NAME=unit-test ]
+      script: make test-travis && ./gradlew codeCoverageReport && bash <(curl -s https://codecov.io/bash)
+
+    - env: [ NAME=crossdock ]
+      script: make crossdock-fresh
+
+    - stage: deploy
+      env: [ NAME=deploy-to-maven ]
+      script: ./gradlew upload
+
+    - env: [ NAME=deploy-to-dockerhub ]
+      script: ./travis/build-crossdock.sh
```

Comment:
not sure I follow this setup. The build shows only two steps now, both from the `test` stage. Where can we see the logs & status from the deploy stage?

![image](https://user-images.githubusercontent.com/3523016/38626102-e449a342-3d79-11e8-9846-36b7ddc68ef2.png)